### ISSUE TITLE
Unexpected errors in playground

### ISSUE THREAD

**Issue by pokrakam** [abaplint/abaplint#3370]
I've experienced several issues with the playground that it seems to give unexpected and often invalid results. I'm guessing some state gets carried on from a previous analysis or something like that? 
It's been hard to reproduce a concrete case as it almost seems random when it happens, but I found one by pasting the following into a new program, deleting the fullstop, pressing enter and adding it back:
```abap
SELECT name1, upper( name1 ) as upper
  FROM ardc
  WHERE length( name ) = 1
  INTO TABLE @DATA(itab)
  UP TO 10 ROWS  ##SUBRC_OK.
```

Behaviour:

https://github.com/user-attachments/assets/7526b410-dc28-42a2-80fc-d5e9e9e257ae

Once errors appear they will stay until the whole block is deleted and added back.

**Comment by larshp** [comment]
looks like there is a bug in one of the rules, 

![image](https://github.com/user-attachments/assets/35e63f18-18d2-4629-876d-32e68765950f)

**Comment by larshp** [comment]
its not very common to not have any code after SELECT :)

**Comment by pokrakam** [comment]
Thanks, the example was left over after process of elimination to try to reproduce the oddities - basically some messages become 'sticky'. Good pointer on using the console when I see suspicious results, will use that for future issues :)

**Comment by larshp** [comment]
@pokrakam try again, new version published
